### PR TITLE
Improve DocuSign integration

### LINKED ISSUE #2515: Get TAR form status back from docusign

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #2515

### CODE REVIEW

File: app/Nova/DocuSignEnvelope.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+            URL::make('DocuSign URL', 'url')
+                ->onlyOnDetail(),
+
+            Text::make('IP Address', 'signer_ip_address')
+                ->onlyOnDetail(),
+
+            Panel::make('Documents', [
+                ...(null === $this->membership_agreement_filename ? [] : [
+                    File::make('Membership Agreement', 'membership_agreement_filename')->disk('local'),
```
Comment (JoshOldenburg) [inline app/Nova/DocuSignEnvelope.php:110]: Do these all need `onlyOnDetail()` to make them not editable?

File: app/Policies/DocuSignEnvelopePolicy.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * Determine whether the user can view the model.
+     */
+    public function view(User $user, DocuSignEnvelope $docuSignEnvelope): bool
+    {
+        return true;
```
Comment (JoshOldenburg) [inline app/Policies/DocuSignEnvelopePolicy.php:28]: Should this restrict to themselves and `can-access-nova`?

File: app/Nova/DocuSignEnvelope.php (hunk 1 above)
Comment (kberzinch) [reply app/Nova/DocuSignEnvelope.php:110]: This model is not editable at all.

File: app/Policies/DocuSignEnvelopePolicy.php (hunk 2 above)
Comment (kberzinch) [reply app/Policies/DocuSignEnvelopePolicy.php:28]: Well, the primary contact and officers need to be able to view it as well. But i can restrict to those groups.

**Review by JoshOldenburg — DISMISSED** [review]

**Review by JoshOldenburg — DISMISSED** [review]

File: app/Jobs/ProcessPostmarkInboundWebhook.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                '/(?P<url>https:\/\/na3.docusign.net\/Member\/EmailStart.aspx.+)/',
+                $payload['TextBody'],
+                'url',
+                'email text'
+            ))->trim();
+
+            Storage::makeDirectory('docusign/'.$envelope->envelope_id);
+
+            collect($payload['Attachments'])->each(static function (array $value, int $key) use ($envelope): void {
+                $original_filename = $value['Name'];
```
Comment (evan10s) [inline app/Jobs/ProcessPostmarkInboundWebhook.php:120]: Is `$value['Name']` guaranteed to exist and be safe to use in a path on disk (e.g., no spaces / not vulnerable to filename attacks)?  I couldn't quite find an answer in the docs

Comment (kberzinch) [reply app/Jobs/ProcessPostmarkInboundWebhook.php:120]: It is (at least theoretically) input by someone with access to configure DocuSign powerforms within apiary, so i think the attack surface is low.

Comment (kberzinch) [reply app/Jobs/ProcessPostmarkInboundWebhook.php]: I added a regex just to be safe.

**Review by evan10s — APPROVED** [review]

**Pull request merged**